lib/commons/reusable_category_icon.dart

```diff
@@ -0,0 +1,45 @@
+import 'package:flutter/material.dart';
+import 'package:furniture/models/items_list.dart';
+
+class CaategoryIcon extends StatefulWidget {
+  const CaategoryIcon(this.index, {super.key});
+  final int index;
+  @override
+  State<CaategoryIcon> createState() => _CaategoryIconState();
+}
+
+class _CaategoryIconState extends State<CaategoryIcon> {
+  @override
+  Widget build(BuildContext context) {
+    bool select = categories[widget.index].select;
```

Comment:
state variables should not be declared in build function
it should be declared above build function